src/main/java/com/avioconsulting/mule/logger/internal/listeners/CustomLoggerAbstractNotificationListener.java

```diff
@@ -41,4 +47,56 @@ protected void logMessage(ComponentLocation location, Event event, String logMes
     logger.log(logProperties, messageAttributes, exceptionProperties, additionalProperties, config,
         location, correlationId);
   }
-}
+
+  protected Map<String, String> getFlowLogAttributes(EnrichedServerNotification notification) {
+    Map<String, String> value = emptyAttributes;
+    FlowLogConfig flowLogConfig;
+    /**
+     * Flow name can contain wildcard (*)
+     * We only look for wildcard either starting of the string or ending of the
+     * string
+     * ex: mq-listener-* will look for all the flows that starts with mq-listener
+     * ex: *-mq-flow will look for all the flows that ends with -mq-flow
+     **/
+    List<Map.Entry<String, FlowLogConfig>> matchedEntries = config.getFlowLogConfigMap().entrySet().stream()
+        .filter(entry -> matchWildcard(entry.getKey(), notification.getResourceIdentifier()))
+        .collect(Collectors.toList());
+    if (!matchedEntries.isEmpty()) {
+      flowLogConfig = matchedEntries.get(0).getValue();
```

Comment:
Only one should match. We can avoid building list by using the `findFirst` - here is an example - 

```suggestion
    Optional<Map.Entry<String, FlowLogConfig>> matchedEntry = config.getFlowLogConfigMap().entrySet().stream()
        .filter(entry -> matchWildcard(entry.getKey(), notification.getResourceIdentifier()))
        .findFirst();
    if (matchedEntry.isPresent()) {
      flowLogConfig = matchedEntry.get().getValue();
```

Reply:
updated